### PR TITLE
fix: syncmeeting made to wait for handleLocusEvent

### PR DESCRIPTION
**Pull request by mkesavan13**
# COMPLETES [SPARK-412189](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-412189)

## This pull request addresses

The ExtCare customer is trying to join the meeting without having a gap for the media to be attached to the locus and that caused both of the users to be stuck in the waiting room.

## by making the following changes

This PR fixes the above issue by waiting for the locus event to be complete and then proceed with syncMeetings. 

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/meetings/index.ts [hunk 1]
```diff
@@ -1082,9 +1088,9 @@ export default class Meetings extends WebexPlugin {
       const activeLocusUrl = [];
 
       if (locusArray?.loci && locusArray.loci.length > 0) {
-        locusArray.loci.forEach((locus) => {
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/meetings/index.ts:1085]: @mkesavan13  do you think we need just this change as there are no aync calls , to do you need all the changes mentioned in the PR ?

Comment (mkesavan13) [reply packages/@webex/plugin-meetings/src/meetings/index.ts:1085]: @arun3528 - You are right. Removed the unwanted changes. Sent build to customer and tested as well. Kindly review again.

**Review by arun3528 — APPROVED** [review]

**Comment by arun3528** [comment]
Looks good , we might have to update the beta branch also

File: packages/@webex/plugin-meetings/src/meetings/index.ts [hunk 2]
```diff
@@ -335,8 +336,13 @@ export default class Meetings extends WebexPlugin {
...
         });
     } else {
-      meeting.locusInfo.parse(meeting, data);
+      return new Promise((resolve) => {
+        meeting.locusInfo.parse(meeting, data);
+        resolve(undefined);
+      });
     }
+
+    return Promise.resolve(undefined);
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meetings/index.ts:345]: as you've added the `async` keyword to the function, then it automatically always returns a promise resolved with whatever return value was returned, so you don't need to convert all the return statements in the function to return a promise - the function `handleLocusEvent()` will always return a promise anyway

File: packages/@webex/plugin-meetings/src/meetings/index.ts [hunk 3]
```diff
@@ -1082,9 +1088,9 @@ export default class Meetings extends WebexPlugin {
       const activeLocusUrl = [];
 
       if (locusArray?.loci && locusArray.loci.length > 0) {
-        locusArray.loci.forEach((locus) => {
+        locusArray.loci.forEach(async (locus) => {
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meetings/index.ts:1091]: `forEach` does not wait for each of the functions invoked on each array element to resolve
see https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

File: packages/@webex/plugin-meetings/src/meetings/index.ts (hunk 2 above)
Comment (mkesavan13) [reply packages/@webex/plugin-meetings/src/meetings/index.ts:345]: Thanks @marcin-bazyl 

The eslint actually threw [consistent-return](https://eslint.org/docs/latest/rules/consistent-return) error. That's why I made this change. Turns out, it accepts `return undefined`

Made this change.

File: packages/@webex/plugin-meetings/src/meetings/index.ts (hunk 3 above)
Comment (mkesavan13) [reply packages/@webex/plugin-meetings/src/meetings/index.ts:1091]: Makes sense @marcin-bazyl 

Refactored it to the `for..of` loop. Thank you.

**Review by marcin-bazyl — APPROVED** [review]
info for others reading this PR: the changes to wait for all handleLocusEvent() calls to resolve in syncMeetings were not actually needed, so in the end the fix is just in selfUtils.ts

**Comment by arun3528** [comment]
> 

@marcin-bazyl  yes the issue will get fixed if we just add a check on the selfUtil but joinedWith should not be null But the root cause was that the promise was not resolved properly which was causing the response parsed after join

**Pull request merged**